backend/vouch/serializers.py

```diff
@@ -8,19 +7,27 @@
 from vouch.models import Voucher
 
 
+# Allow to keep the JSON representation of `GivenVoucherSerializer` and
+# `ReadOnlyVoucherSerializer` similar.
+VOUCHER_FIELDS = ["by", "to", "is_public", "value"]
+
+
 class GivenVoucherSerializer(ModelSerializer):
-    by = serializers.HiddenField(default=serializers.CurrentUserDefault())
+    """
+    A voucher given by the logged-in user to a target user.
+
+    As the public status of the voucher is displayed, this serializer
+    shouldn't be used to list the vouchers not related to the logged-in user.
+    Prefer `PublicGivenVoucherSerializer` instead.
+    """
+
+    by = serializers.CharField(default=serializers.CurrentUserDefault())
     to = serializers.CharField()
 
     class Meta:
         model = Voucher
-
-        fields = [
-            "by",
-            "to",
-            "is_public",
-            "value",
-        ]
+        fields = VOUCHER_FIELDS
+        read_only_fields = ["by"]
```

Comment:
This `read_only_fields` has no effect, as `by` is defined explicitly as a (writable)  `CharField`.  
In particular, this is visible in the API docs: "by" is writable (even if any non-empty value would trigger a server error as a User object is expected to create the Vouch instance).

An alternative solution, with 2 distinct fields for read and write, could make the API schema cleaner:
```py
class GivenVoucherSerializer(ModelSerializer):
    _by = serializers.HiddenField(
        source="by", default=serializers.CurrentUserDefault(), write_only=True
    )
    by = serializers.CharField(read_only=True)
    to = serializers.CharField()

    class Meta:
        model = Voucher
        fields = VOUCHER_FIELDS + ["_by"]
```

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1091/commits/db441b13c030fee368bb4f8d9d784619376e2918.